js/apps/system/_admin/aardvark/APP/frontend/js/collections/arangoDocument.js

```diff
@@ -69,26 +73,34 @@ window.ArangoDocument = Backbone.Collection.extend({
     });
   },
 
-  createTypeDocument: function (collectionID, key, callback, returnNew,
+  createTypeDocument: function (collectionID, key, body, callback, returnNew,
                                 smartJoinAttribute, smartJoinAttributeValue,
                                 smartGraphAttribute, smartGraphAttributeValue) {
-    var newDocument = {};
-
-    if (smartJoinAttribute && smartJoinAttributeValue && key) {
-      // case: smartJoin, bot value are needed and NOT optional
-      newDocument._key = smartJoinAttributeValue + ':' + key;
-      newDocument[smartJoinAttribute] = smartJoinAttributeValue;
-    } else if (smartGraphAttribute && smartGraphAttributeValue) {
-      // case: smartGraph with value
-      // other to smartJoin, we can:
-      // 1.) Create without smartGraphAttribute and without smartGraphAttributeValue
-      // 2.) Create only with smartGraphAttributeValue
-      if (key) {
-        newDocument._key = smartGraphAttributeValue + ':' + key;
+    var newDocument = body;
+
+    try {
```

Comment:
why does this previously existing code now need `try...catch`?

Reply:
if newDocument / body now isn't an object, this will throw. The If is so complicated, I decided to just invoke the function again with an empty object.